src/RedisInterface/Controllers/RobotController.cs

```diff
@@ -375,6 +378,35 @@ public async Task<ActionResult<List<CloudModel>>> GetConnectedCloudsIdsAsync(Gui
             }
 
         }
+        [HttpGet]
+        [Route("{robotId}/topic", Name = "RobotChangeTopicStatus")]
+        [ProducesResponseType(typeof(RosTopicModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ApiResponse), (int)HttpStatusCode.NotFound)]
+        [ProducesResponseType(typeof(ApiResponse), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ApiResponse), (int)HttpStatusCode.InternalServerError)]
+        public async Task<ActionResult> EnableTopicAsync(Guid robotId,string topicName, bool topicEnabled)
+        {
+            try
+            {
+                RobotModel robot = await _robotRepository.GetByIdAsync(robotId);
+                RosTopicModel topicModel = robot.GetTopicModelFromRobot(topicName);
```

Comment:
You have forgot to save the changes to the robot definition after updating topic status

Reply:
save in redis changes with: await _robotRepository.AddAsync(robot, () => robot.Id);

Reply:
The `await _robotRepository.AddAsync(robot, () => robot.Id);` can be moved outside of the `if` statement. It will be executed either way :)